### DISCUSSION TITLE
Some questions as Im not really understanding how to replace my current nginx php fpm setup

### DISCUSSION THREAD

**Discussion by brycelarge** [php/frankenphp#775] [Q&A]
Hi there,

Thanks for the amazing work, very excited about frankenphp.

I have multiple laravel apps on my server and I have setup the Caddyfile and get all the sites working but now when I want to switch on laravel octane. It seems the frankenphp docker can only handle 1 laravel app and thats done by building your own modification of the docker just to change the ENTRYPOINT to ["php", "artisan", "octane:start"]. Is there something I am missing or is this by desgin that only 1 app per docker? Would be great if the documentation could have an example for multiple laravel apps with octane running.

Im also a little confused about how to use laravel within the frankenphp docker. It would help if the laravel documentation could be updated to include some more basic information:
1. is the worker used for the laravel scheduler or how would you setup a traditional cron to run laravel scheduler every minute?
2. Running artisan commands
3. Changing php.ini for artisan commands vs browser php.ini

Apologies if these questions have been asked before, I did go through all the existing issues but could not find anything that helped me.

Many thanks
Bryce

**Comment by withinboredom** [comment]
These seem to be Laravel specific questions. That's like going to nginx/apache and asking Laravel specific questions. We can try to help you, but you'll get better answers from Laravel...

> Is there something I am missing or is this by desgin that only 1 app per docker?

This is a Docker philosophy that only one process should run in a container at a time. I believe Laravel breaks this philosophy, so Laravel would need to do something custom here.

**Comment by lukevi** [comment]
Sorry this is a bit late to this discussion, but I'm just running into the same thing now that I'm trying out FrankenPHP. I can attest that it's not Laravel specific - many php apps use cron jobs and async job queues, with or without a framework. With plain php containers, I usually run a container for each process, to follow the Docker philosophy as you mention. One runs cron, one runs jobs off of the queue, and the main one serves web requests.

Perhaps the queue worker at least could be set up as a second container in docker Compose, like:
```
  php:
    image: dunglas/frankenphp
    # ...

  queue-worker:
    image: dunglas/frankenphp
    env:
      FRANKENPHP_CONFIG: "worker workerscript.php"
```

Thoughts?
